src/Indigo/Contracts/HomebaseLite/Types.hs

```diff
@@ -0,0 +1,67 @@
+-- SPDX-FileCopyrightText: 2022 Tezos Commons
+-- SPDX-License-Identifier: LicenseRef-MIT-TC
+
+{-# OPTIONS -Wno-orphans #-}
+
+module Indigo.Contracts.HomebaseLite.Types
+  ( Parameter(..)
+  , Storage(..)
+  , Configuration(..)
+  , URI(..)
+  ) where
+
+import Indigo
+
+data Parameter
+  = Set_admin KeyHash
+  | Accept_admin ()
+  | Add_maintainers [KeyHash]
+  | Remove_maintainers [KeyHash]
+  | Configure Configuration
+  | Propose (URI, [MText])
+  | Vote (URI, Natural)
```

Comment:
Arguably a newtype for this `Natural` here would be useful as well, given that at a first glance it's not really clear that this denotes a choice.

Alternatively (I actually recommend this), you could make a used `Named` here for each side of the tuple (as well as for the argument of `Propose`) and make a type synonym as well for convenience.